survey_formio/models/survey.py

```diff
@@ -84,25 +78,16 @@ def _from_formio_multiple_choice_to_odoo(
           "answer_2_value": true/false
           ....
         }}
-        Odoo expects the following format:
-        {"question_id_answer_id": "answer_id"}
+        Odoo expects the following format: [answer_id, answer_id, ...]
         Form.io returns all the answers with true or false but odoo doesn't
         distinguish them like that. If the answer isn't selected, it shouldn't appear
-        in the dictionnary containing the answers.
-
-        Returns a dictionnary with the correct format.
-
+        in the list containing the answers.
+        Returns a list with the correct format.
         Parameters:
         question_id: id of the survey.question
         multiple_choice_answers_dict: dict given by form.io for a multiple choice question
         """
-        answers_dict = {}
-        multiple_choice_answers_dict = {
-            key: value for (key, value) in multiple_choice_answers_dict.items() if value
-        }
-        for answer in multiple_choice_answers_dict.keys():
-            answers_dict[question_id + "_" + answer[1:]] = answer[1:]
-        return answers_dict
+        return [value for value in multiple_choice_answers_dict.values() if value]
```

Comment:
I've tested this and it seems there's an issue here. 
Saving a multiple choice answer given by formio doesn't work.  I think Odoo is waiting for the id of the answer, not its value (see : https://github.com/odoo/odoo/blob/16.0/addons/survey/tests/test_survey_ui_session.py#L210 ) 
I think this should be 
`return [value for value in multiple_choice_answers_dict.keys() if value]` or something like that

Reply:
Should be better now

Reply:
It works now. Thanks :)

Reply:
Thanks for the review!